kernel/src/mem/phys.rs

```diff
@@ -55,15 +263,95 @@ impl PhysicalMemory {
         allocator().lock().allocate_frames(n)
     }
 
-    /// Calls [`FrameAllocator::deallocate_frame`] on the current physical allocator.
+    /// Deallocates a single physical frame, returning it to the free pool.
+    ///
+    /// This method marks the frame as free, making it available for future allocations.
+    /// The frame must have been previously allocated with [`allocate_frame()`](Self::allocate_frame).
+    ///
+    /// # Type Parameters
```

Comment:
Discard the type parameter sections. The compiler enforces that, and the IDE helps showing implementations.